### PR TITLE
input-fonts: make Nix download the source

### PR DESCRIPTION
**Pull request by jtojnar**
###### Motivation for this change
The website generates a ZIP archive with fresh TTF files for each download, which will have different “modified” field in the TTF metadata each time. This makes `requireFile` useless as the ZIP file will not have a static hash.

Instead, let’s make user accept the license in Nix and download the file for them.  Then, we can post-process it and hopefully achieve a somewhat fixed output.

This is still not really reproducible since:

- the font can be updated (last update in 2015)
- the fonttools used by the server can be updated to one producing a different output
- the fonttools used by this package can be updated
- the fonttools might actually be non-deterministic

But hopefully these events are rare so it will be more stable than the ZIP produced by upstream, which changes every time. When that happens, we can always just update it like we did before. We do not need to worry about cache since the package is unfree.

I also added myself as a maintainer.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/fonts/input-fonts/default.nix [hunk 1]
```diff
@@ -1,31 +1,76 @@
...
+    Input is available free of charge for private/unpublished usage. This includes things like your personal coding app or for composing plain text documents.
+    To use it, you need to agree to its license: https://input.djr.com/license/
+
+    You can express acceptance by setting acceptLicense to true in your
+    configuration. Note that this is not a free license so it requires allowing
+    unfree licenses.
+
+    configuration.nix:
+      nixpkgs.config.allowUnfree = true;
+      nixpkgs.config.input-fonts.acceptLicense = true;
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/input-fonts/default.nix:21]: First time I see that you need to explicitly accept a license. Isn't allowUnfree enough?

**Review by romildo — APPROVED** [review]
Tested locally. Looks good to me.

**Comment by vcunat** [comment]
So this needs picking to stable because otherwise it can't be installed, right?  ([example](https://unix.stackexchange.com/questions/644439/issue-installing-input-fonts-with-nixos-how-to-use-nix-store-add-fixed))

**Comment by jtojnar** [comment]
Yes, if someone wants to use Input fonts on stable comfortably, they will
need to backport it.

**Pull request merged**